bitcoin/examples/ecdsa-psbt-simple.rs

```diff
@@ -77,7 +77,7 @@ fn get_internal_address_xpriv<C: Signing>(
         BIP84_DERIVATION_PATH.into_derivation_path().expect("valid derivation path");
     let child_xpriv = master_xpriv.derive_priv(secp, &derivation_path);
     let internal_index = ChildNumber::ONE_NORMAL;
-    let idx = ChildNumber::from_normal_idx(index).expect("valid index number");
+    let idx = ChildNumber::from_normal_number(index).expect("valid index number");
```

Comment:
Actually addressing some comments of the old PR

Reply:
Nope, it needs to be `index`. There was confusion that `number` and `index` are same but they are actually different things.

Reply:
Are we going with `index` or `number`? In the original draft PR (

@apoelstra mentioned: https://github.com/rust-bitcoin/rust-bitcoin/pull/2911#issuecomment-2186672422

> This looks great to me. A minor point on naming is that I notice we have `from_*_idx` methods, which are weird because (a) `idx` is an abbreviation that we don't use in other method names (according to our API docs it does appear in the `bip152` module which we should also fix), but (b) our types are called `Number`s not indexes.
> 
> I think we should change the names of the new types to `HardenedChildNumber` etc rather than just `HardenedChild`. This is a mouthful but I think usually users won't be using these, since in almost all cases they'd want the more-generic-and-shorter `ChildNumber` enum. **And** we should rename `from_hardened_idx` to `from_hardened_child_number` and same for the non-hardened one.

Reply:
That was a mistake which was caught here: https://github.com/rust-bitcoin/rust-bitcoin/pull/2911#discussion_r1652345786

Reply:
Yeah, thanks for clarifying. I agree let's keep `index` now.